build/tsc.gni

```diff
@@ -0,0 +1,40 @@
+import("npm.gni")
+
+template("typescript_build") {
+  assert(defined(invoker.tsconfig),
+         "Need tsconfig name to run (must exist in electron root)")
+  assert(defined(invoker.sources), "Need tsc sources to run")
+  assert(defined(invoker.electron_dir),
+         "Need electron_dir to run, should be 'lib' or other top level dir")
+
+  npm_action(target_name) {
+    forward_variables_from(invoker,
+                           [
+                             "deps",
+                             "public_deps",
+                             "outputs",
+                           ])
+    script = "tsc"
+
+    sources = invoker.sources
+
+    args = [
+      "-p",
+      "tsconfig." + invoker.tsconfig + ".json",
+      "--outDir",
+      rebase_path("$target_gen_dir/tsc/electron/" + invoker.electron_dir),
+      "--typeRoots",
+      "node_modules/@types,typings," +
+          rebase_path("$target_gen_dir/tsc/electron/typings"),
```

Comment:
what generates `tsc/electron/typings`?

Reply:
This implies a dependency on `tsc/electron/typings`. Either state that dependency here, or expose the `typeRoots` functionality in such a way that this template can function without that implicit dependency.